components/camera/platforms/jetson/camera.go

```diff
@@ -0,0 +1,139 @@
+	board, ok := cameraInfoMappings[osInfo.Device]
+	if !ok {
+		return fmt.Errorf("the %s device is not supported on this platform", osInfo.Device)
+	}
+	daughterboard, ok := board.Daughterboards[daughterboardName]
+	if !ok {
+		return fmt.Errorf("the %s daughterboard is not supported on this platform", daughterboardName)
+	}
+	driver, ok := board.Modules[driverName]
+	if !ok {
+		return fmt.Errorf("the %s driver is not supported on this platform", driverName)
+	}
+	if err := checkDaughterBoardConnected(daughterboard); err != nil {
+		return fmt.Errorf("the %s daughterboard is not connected or not powerd on."+
+			"Please check daughter-board connection to the %s",
+			daughterboardName, osInfo.Device)
+	}
+	if err := checkDriverInstalled(osInfo.Kernel, driver); err != nil {
+		return fmt.Errorf("the %s driver not installed. Please follow instructions for driver installation", driverName)
+	}
+
+	return fmt.Errorf("the %s daughterboard is connected and "+
+		"%s camera driver is installed on the %s."+
+		"please check that the video path is correct and driver is working",
+		daughterboardName, driverName, osInfo.Device)
+}
+
+// checkDaughterBoardConnected checks if the daughterboard is connected
+// by looking for the I2C bus interfaces associated with the daughterboard
+func checkDaughterBoardConnected(daughterboard []string) error {
+	for _, i2c := range daughterboard {
+		err := checkI2CInterface(i2c)
+		if err != nil {
+			return fmt.Errorf("unable to verify that daughterboard is connected: %w", err)
+		}
+	}
+	return nil
+}
+
+// checkI2CInterface checks if the I2C bus is available
+func checkI2CInterface(bus string) error {
+	i2cPath := filepath.Join("/dev", bus)
+	if err := checkFileExists(i2cPath); err != nil {
+		return fmt.Errorf("unable to verify that i2c bus is available: %w", err)
+	}
+	return nil
+}
+
+// checkDriverInstalled checks if the driver is installed for the
+// given kernel version and object file target
+func checkDriverInstalled(kernel, driver string) error {
+	driverPath := filepath.Join("/lib/modules", kernel, "extra", driver)
+	if err := checkFileExists(driverPath); err != nil {
+		return fmt.Errorf("unable to verify that camera driver is installed: %w", err)
+	}
+	return nil
+}
+
+// checkFileExists is a helper function that wraps os.Stat
+// errors are parsed to return a more specific error message
+func checkFileExists(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return fmt.Errorf("%s does not exist", path)
+	} else if err != nil {
+		switch {
+		case os.IsPermission(err):
+			return fmt.Errorf("permission denied when trying to access %s", path)
+		case errors.Is(err, syscall.ENOSPC):
+			return fmt.Errorf("device is out of space. unable to check %s", path)
+		case errors.Is(err, syscall.ENOMEM):
+			return fmt.Errorf("device is out of memory. unable to check %s", path)
+		default:
+			return fmt.Errorf("unable to access %s", path)
```

Comment:
I appreciate that you've tried to make this thorough, but wouldn't it be both simpler and more useful to just return `err` no matter what? All the things you return hide the original error, and we really only care about whether we can find the file, right?

```go
func checkFileExists(path string) error {
    _, err := os.Stat(path)
    if err == nil {
        return nil
    }
    return fmt.Errorf("unable to stat %s: %w", path, err)
}
```

and even that might be overkill; perhaps we just return `err` without any additional formatting.

Reply:
Totally agree - simplified the file check.